### ISSUE TITLE
Normative Canonicalisation format

### ISSUE THREAD

**Issue by mlagally** [w3c/wot-profile#132]
We need a normative canonicalisation format in the profile specification. This depends on the testability, i.e. 2 implementations.
The TD should make a definition the profile should reference it.

**Comment by benfrancis** [comment]
> We need a normative canonicalisation format in the profile specification.

Since the current WoT Profile 1.0 draft does not use canonicalisation (previous drafts did, but those references have now been removed), are there features you think are currently missing from the specification which would rely on canonicalisation?

> The TD should make a definition the profile should reference it.

See section [6.6 Canonicalization](https://w3c.github.io/wot-thing-description/#canonicalization-serialization-json) of the WoT Thing Description 1.1 specification.

**Comment by mlagally** [comment]
Canonical TDs help to identify if two different TDs describe the same thing. 
Determining identity is a simple string comparison.

Usually there are globally unique IDs for that purpose, 
however during the publication of the 1.0 TD specification
 there were concerns raised that a unique ID may cause privacy issues.

**Comment by egekorkan** [comment]
I agree that canonicalisation is important but how does it help the interoperability claim of the core profile? I think how it is used should made more clear

**Comment by mlagally** [comment]
@ege, @benfrancis : without canonicalisation and a unique id, a consumer is not able to check identity of 2 TDs that describe the same thing.

**Comment by egekorkan** [comment]
But why should I do that in the first place a profile compliant consumer? What is the use case?

**Comment by mlagally** [comment]
arch call on 25.11.
The term "canonicalisation" is no longer appropriate, however the TD section on canonicalisation contains several constraints.
These constraints should be moved out of the TD and constraints are considered for inclusion in the profile.
Canonical format is not a requirement at this point.

**Comment by mlagally** [comment]
After the TD has defined a canonicalisation format, Profile 2.0 will adopt it.
